vizro-core/src/vizro/actions/_actions_utils.py

```diff
@@ -192,12 +200,12 @@ def _get_modified_page_charts(
         parameters=ctds_parameters,
     )
 
-    outputs = {
-        target: model_manager[target](  # type: ignore[index, operator]
-            data_frame=filtered_data[target],
-            **parameterized_config[target],
-        ).update_layout(template="vizro_dark" if ctd_theme["value"] else "vizro_light")
-        for target in targets
-    }
+    outputs = {}
+    for target in targets:
+        outputs[target] = model_manager[target](  # type: ignore[operator]
+            data_frame=filtered_data[target], **parameterized_config[target]
+        )
+        if hasattr(outputs[target], "update_layout"):
```

Comment:
Not sure I like this, but it's probably ok, beyond the scope of this PR anyway and is probably just I don't yet understand how the themes and actions currently work...

But in short, why do we need to `update_layout` here as well as the `_update_theme` callback you moved to `Graph`?

Reply:
Although, the `_update_theme` callback (that is moved to the Graph) is triggered when users change the theme on the toggle button, we still need to apply `update_layout` to graph when the filter/parameter/filter_interaction action is triggered in purpose to apply the currently selected theme (as template) to the graph that will be returned as a response. 

Try to comment out the following code:
```
if hasattr(outputs[target], "update_layout"):
    outputs[target].update_layout(template="vizro_dark" if ctd_theme["value"] else "vizro_light")
```
then open the app, select light theme (on toggle button), and apply the filter which has Graph target. The graph will be presented in a dark theme, in contrast to the rest of the page, which is in a light theme.

Reply:
Thanks for the explanation! I'd like to have a think about whether there's a nicer way to do this but it seems fine for now 👍